presto-main/src/main/java/com/facebook/presto/operator/scalar/MapToMapCast.java

```diff
@@ -274,4 +283,15 @@ public static Block mapCast(
         mapBlockBuilder.closeEntry();
         return (Block) toMapType.getObject(mapBlockBuilder, mapBlockBuilder.getPositionCount() - 1);
     }
+
+    public static boolean canSkipDuplicateKeysCheck(Type castFromType, Type castToType)
```

Comment:
instead of special-casing can we introduce the following utility functions if these aren't already in the code base:
```
isUpcast(fromType, toType)
isLosslessCast(fromType, toType) // lossless cast is either an upcast or some combination of types e.g. integer->string
```

Reply:
Should the new isUpcast and isLosslessCast be comprehensive? If yes, it will be notrival to include all possible conversions and not worth the effort considering Presstisimo is coming soon.
I am specializing for the int -> bigint/varchar and bigint -> varchar here, which is the most common cases for map cast. 
A compromise is to add the conversion cases we are interested in these isUpcast and isLosslessCast util functions, however with comment saying that this is not a comprehensive list and returning false does not necessarily mean that it is not upcast/lossless cast, which I am not sure if it's really helpful.

Reply:
A better task would be to revisit/fix implicit parameter conversions/casts for map operations for all types in the analyzer.